test/lib/pgBackRestTest/Module/Manifest/ManifestAllTest.pm

```diff
@@ -0,0 +1,1088 @@
+            MANIFEST_SUBKEY_SIZE, length($strTestNew));
+        $oManifestExpected->set(MANIFEST_SECTION_TARGET_FILE, MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_TIMESTAMP, $lTime);
+
+        $oLastManifest->set(MANIFEST_SECTION_TARGET_FILE, MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_SIZE, length($strTestNew));
+        $oLastManifest->set(MANIFEST_SECTION_TARGET_FILE, MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_TIMESTAMP, $lTime);
+
+        # Set a reference, checksum, repo size, master and page checksum in the last manifest
+        my $strCheckSum = '1234567890';
+        my $lRepoSize = 10000;
+        $oLastManifest->set(MANIFEST_SECTION_TARGET_FILE, MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_REFERENCE, BOGUS . BOGUS);
+        $oLastManifest->set(MANIFEST_SECTION_TARGET_FILE, MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_CHECKSUM, $strCheckSum);
+        $oLastManifest->numericSet(MANIFEST_SECTION_TARGET_FILE, MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_REPO_SIZE, $lRepoSize);
+        $oLastManifest->boolSet(MANIFEST_SECTION_TARGET_FILE,  MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_MASTER, false);
+        $oLastManifest->boolSet(MANIFEST_SECTION_TARGET_FILE,  MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_CHECKSUM_PAGE, true);
+
+        # Update expected manifest
+        $oManifestExpected->set(MANIFEST_SECTION_TARGET_FILE, MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_REFERENCE, BOGUS . BOGUS);
+        $oManifestExpected->set(MANIFEST_SECTION_TARGET_FILE, MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_CHECKSUM, $strCheckSum);
+        $oManifestExpected->numericSet(MANIFEST_SECTION_TARGET_FILE, MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_REPO_SIZE, $lRepoSize);
+        $oManifestExpected->boolSet(MANIFEST_SECTION_TARGET_FILE,  MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_CHECKSUM_PAGE, true);
+
+        # Default "master" is flipping because it's not something we read from disk
+        $oManifestExpected->boolSet(MANIFEST_SECTION_TARGET_FILE . ":default", MANIFEST_SUBKEY_MASTER, undef, false);
+        $oManifestExpected->boolSet(MANIFEST_SECTION_TARGET_FILE,  MANIFEST_TARGET_PGDATA . '/' . $strTest,
+            MANIFEST_SUBKEY_MASTER, true);
+
+        $oManifest->build(storageDb(), $self->{strDbPath}, $oLastManifest, true);
+        $self->testResult(sub {$self->manifestCompare($oManifestExpected, $oManifest)}, "", 'updates from last manifest');
+
+# CSHANG What is the [backup:option] option-checksum-page setting? Why are the checksum-page/error settings per file not dependent on this? What are these for?
```

Comment:
This determines whether page checksums will be done or not.  The checksum-page/error settings are dependent on it.